azurelinuxagent/common/event.py

```diff
@@ -235,14 +234,13 @@ def save_event(self, data, prevent_telemetry_logging=True):
             raise EventError(msg)
 
         existing_events = os.listdir(self.event_dir)
-        if len(existing_events) >= 1000:
-            # To prevent the chance of #1035 happening - introduced periodic_WARNs.
-            # More details about it - azurelinuxagent.common.event.add_log_event's pydoc.
-            logger.warn("Too many files under: {0}, current count:  {1}, removing oldest".format(
-                                     self.event_dir, len(existing_events)))
-            existing_events.sort()
-            oldest_files = existing_events[:-999]
+        if len(existing_events) >= MAX_NUMBER_OF_EVENTS:
+            logger.periodic_warn(logger.EVERY_MINUTE, "[PERIODIC] Too many files under: {0}, current count:  {1}, "
+                                                      "removing oldest".format(self.event_dir, len(existing_events)))
             try:
+                existing_events = os.listdir(self.event_dir)
```

Comment:
NIT: `existing_events` already populated in line 236, line 241 is duplicated and not required

Reply:
Explained above (https://github.com/Azure/WALinuxAgent/pull/1659#discussion_r348859779)